### DISCUSSION TITLE
Create built-in Laravel unique_with validation rule

### DISCUSSION THREAD

**Discussion by MatteoT92** [laravel/framework#53245] [Ideas]
Hi everyone.
I ask you if is possible implementing an rule built-in as actual unique rule but applied on composite key (unique or primary)?

**Comment by shaedrich** [comment]
You could make the `$column` property of `\Illuminate\Validation\Rules\DatabaseRule` plural and an `array` or `string|string[]`:
https://github.com/laravel/framework/blob/6bed63f07c3b9ffc187fa3161cf116c43d6b41eb/src/Illuminate/Validation/Rules/DatabaseRule.php#L48-L53

Technically, the `\Illuminate\Validation\Rules\Unique::ignoreModel()` method could take case of that automatically:
https://github.com/laravel/framework/blob/6bed63f07c3b9ffc187fa3161cf116c43d6b41eb/src/Illuminate/Validation/Rules/Unique.php#L53-L59
but as the docs state, Laravel doesn't support composite keys natively, so I'm not surprise, that it doesn't. 

> Eloquent requires each model to have at least one uniquely identifying "ID" that can serve as its primary key. "Composite" primary keys are not supported by Eloquent models. However, you are free to add additional multi-column, unique indexes to your database tables in addition to the table's uniquely identifying primary key.

It's already possible to use `\Illuminate\Validation\Rules\Unique::using()`:
```php
use Illuminate\Validation\Rule;

class MyRequest extends FormRequest
{
    public function rules(): array
        return [
            'user_id' => [ Rule::unique(User::class, 'id')->using(function($query) use($user) {
                $query->where([ 'id' => $this->user_id, 'name' => $this->user_name ]);
            }) ],
        ];
    }
}
```
You might [customize the error message](https://laravel.com/docs/11.x/validation#customizing-the-error-messages) in that case, which is "The user id has already been taken."

But the thing is: The Laravel validation rules are built to validate each field separately, so this would break a bit with that convention.
